lib/webpacker/commands.rb

```diff
@@ -7,10 +7,11 @@ def initialize(webpacker)
 
   def clean(count_to_keep = 2)
     if config.public_output_path.exist? && config.public_manifest_path.exist?
-      files_in_manifest = manifest.refresh.values.map { |f| File.join config.root_path, "public", f }
-      files_to_be_removed = files_in_manifest.flat_map do |file_in_manifest|
+      files_in_manifest = process_manifest_hash(manifest.refresh)
+      files_to_be_removed = files_in_manifest.map do |file_in_manifest|
         file_prefix, file_ext = file_in_manifest.scan(/(.*)[0-9a-f]{20}(.*)/).first
         versions_of_file = Dir.glob("#{file_prefix}*#{file_ext}").grep(/#{file_prefix}[0-9a-f]{20}#{file_ext}/)
+        next "" unless versions_of_file.any?
```

Comment:
```suggestion
        next unless versions_of_file.any?
```
I'm not sure why it's preferable to insert `""` instead of `nil`

Reply:
Actually, on second thought, I think it's better to just remove this line. You end up with an array like `[nil, nil, nil]` if there are no versions of 3 files instead of `[]` with the previous code. Your version will require the last line of this method to be changed to `files_to_be_removed.compact.each { |f| File.delete f }` (added `.compact`) in order to avoid `TypeError: no implicit conversion of nil into String`